pennylane_forest/qpu.py

```diff
@@ -82,6 +82,14 @@ def __init__(self, device, *, shots=1024, active_reset=True, load_qc=True, reado
         self.readout_error = readout_error
 
         self._eigs = {}
+        self.parametric_compilation = kwargs.get("parametric_compilation", True)
+
+        if self.parametric_compilation:
+            self._lookup_table = {}
```

Comment:
It's not clear to me where `_lookup_table` is used. And unlike the other two, it does not seem to be defined in `device.py`

Reply:
There are two subtle things to consider here:
*  ``QPUDevice`` inherits the ``generate_samples`` method from ``QVMDevice``. This method uses ``_lookup_table`` to store compiled programs.
* Currently, ``QPUDevice__init__`` calls on the ``ForestDevice.__init__`` method and ``QVMDevice.__init__`` is not called, (coming from  [``super(QVMDevice, self).__init__(num_wires, shots, **kwargs)``](https://github.com/rigetti/pennylane-forest/blob/9f104463d4a101d09082952f1377236b0759da6e/pennylane_forest/qpu.py#L103)). As both the ``QPUDevice`` and ``QVMDevice``, require the dictionaries, they initialize them separately. This design and how the ``__init__`` methods are called could be altered in a subsequent PR.

As for the method ``generate_samples``, this is called in ``QubitDevice.execute`` (and thus for every device defined in this plugin).